src/Forms/FieldsValidator.php

```diff
@@ -13,7 +13,9 @@ public function php($data): bool
         $fields = $this->form->Fields();
 
         foreach ($fields as $field) {
-            $valid = ($field->validate($this) && $valid);
+            $result = $field->validate();
+            $valid = $result->isValid() && $valid;
+            $this->result->combineAnd($result);
         }
 
         return $valid;
```

Comment:
```suggestion
            $this->result->combineAnd($field->validate());
        }
        return $this->result->isValid();
```
And remove the `$valid = true` at the beginning. Much more succinct and doesn't have the unnecessary variable.

Reply:
Updated